### ISSUE TITLE
Ability to recover individual URLs

### ISSUE THREAD

**Issue by probablykasper** [ninsky/recovermyvideotracker#2]
I assume recovermy.video has some database of URLs and their titles? If so, this should be pretty possible in case you have videos you want to recover, and haven't used recovermy.video before.

**Comment by ninsky** [comment]
RecoverMy.Video just covers the scope of your own playlist videos. Could you give an example of such a case where you would use an individual URL? 

(Wasn't this feature request related to the "liked videos" playlist which can't be monitored right now? If so, I would propose to rename the request to e.g. "Add monitoring of liked videos playlist")

**Comment by probablykasper** [comment]
@ninsky If you haven't used RecoverMy.Video before, you might have a playlist with a deleted video. So, what you would do then is to type the URL into RecoverMy.Video in case it's been recovered by somebody already. You could do it if you find deleted videos in others' playlists as well.

Note that what I'm talking about would be video URLs, not playlist URLs.

The liked videos thing is separate. I can make a new issue for it if you'd like.

**Comment by ninsky** [comment]
> So, what you would do then is to type the URL into RecoverMy.Video in case it's been recovered by somebody already

Ah, okay. Now I've got your point. So it would be some kind of manual look up if someone else on RecoverMy.Video recovered this video in the past. This would **maybe** work for some kind of popular videos... I'm not sure how likely a match would be as there are users from all over the world on RecoverMy.Video. On the other side, this should not be a big deal to implement. It's something like Archive.org already does.... Let me put this on my backlog with prio=medium/low, as long as this is not asked for by other users.

>The liked videos thing is separate. I can make a new issue for it if you'd like.

Would be nice, thanks

**Comment by Berny23** [comment]
Hey @ninsky & @SpectralKH, this feature would be very useful!
I have 60 completely "Lost" videos and don't know the titles. But with a simple database search function (maybe just for registered users and not for everyone) I could type in my lost video's urls and find some of the more popular ones. It would also be nice to have a button next to "Archive.org" for this. :)

**Comment by TheTyrius** [comment]
This is something I was wondering as well. This could also be done when a user first registers (if it isn't already?). I can imagine that searching the "global" database (vs just your previous recorded data) might be slow/computationally expensive, in that case, it might be possible to put such urls in a queue, and process them whenever there is low load overall? 

Still the question remains how effective this approach would be (directly related to the amount of url/title combos the service can aggregate).

**Comment by probablykasper** [comment]
@TheTyrius  Yeah, essentially archive.org but instead of saving the entire page, just saving essential metadata, which allows a lot more videos to be saved easily. You could probably save channel urls, names, descriptions and thumbnails too. I'd love to see view counts saved as well!

Why exactly would it be slow/computationally expensive?

**Comment by Antoine-Patel** [comment]
> as long as this is not asked for by other users

I'm in the same situation (search engines and archive.org are no help), had the same idea, and found this issue. 

Although it might be unlikely to find a match, I'd rather have a "not found" answer to know that it's hopeless in order to move on, than being stuck with the thought that the answer might be sitting in a database I don't have (sufficient) access to.

The video was too recent and short-lived for me to hope finding it in a random youtube dataset, so it's my best (and last) bet at this point.

All this drama to basically say that I'm really looking forward to this feature. Even rate limiting this feature to one manual query per day to avoid overloading the database(s) would be nice to have.

**Comment by ninsky** [comment]
Okay, I see your pain ;) In many cases YouTube removes playlist items and just shows a message to users that "some items have been removed". In such a case you don't have any reference to those videos. But there are also cases where you get displayed a "deleted video" item on your playlist. 

Which information would you like to use to search for a deleted video if such a database lookup would exist?

**Comment by Antoine-Patel** [comment]
Thanks for your answer !

> Which information would you like to use to search for a deleted video if such a database lookup would exist?

I see what you mean; thankfully in my case I still have the reference to the video in the playlist itself: https://www.youtube.com/watch?v=ASghU1J6Ltg. The user should at least have the **`URL or the youtube video ID`** to begin with in my opinion.

Ideally the input field in the UI would take the URL of the video, then the code would extract the video ID (ASghU1J6Ltg) from it, and run the actual lookup with that. 

Alternatively you could ask the users to provide the video ID directly, but from a UI/UX perspective it's a bit less user-friendly. Yet another way would be to detect whether the input is a video URL or a video ID, and allow both. It's up to you to decide, any of those options would be fine.

Bonus: could be relatively easy to create, if not already in place, an index <sup>(*)</sup> on the video ID column/field, for efficient lookups, assuming the database technology supports that.

<sup>(*) assumption here: relational or noSQL db, I might be off the mark</sup>
